kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconcilerTest.java

```diff
@@ -54,39 +57,116 @@ class KafkaProtocolFilterReconcilerTest {
             .endSpec()
             .build();
 
+    public static final KafkaProtocolFilter FILTER_MULTIPLE = new KafkaProtocolFilterBuilder()
+            .withNewMetadata()
+                .withName("foo")
+                .withGeneration(42L)
+                .withUid("foo-uid")
+            .endMetadata()
+            .withNewSpec()
+                .withType("org.example.MyFilter")
+                .withConfigTemplate(Map.of(
+                        "normalProp", "normalValue",
+                        "securePropA", "${secret:my-secret:key}",
+                        "securePropB", "${secret:my-secret2:key}",
+                        "securePropC", "${configmap:my-configmap:key}",
+                        "securePropD", "${configmap:my-configmap2:key}"))
+            .endSpec()
+            .build();
+
+    public static final KafkaProtocolFilter FILTER_SECRET_ONLY = new KafkaProtocolFilterBuilder()
+            .withNewMetadata()
+            .withName("foo")
+            .withGeneration(42L)
+            .withUid("foo-uid")
+            .endMetadata()
+            .withNewSpec()
+            .withType("org.example.MyFilter")
+            .withConfigTemplate(Map.of(
+                    "normalProp", "normalValue",
+                    "securePropA", "${secret:my-secret:key}"))
+            .endSpec()
+            .build();
+
+    public static final KafkaProtocolFilter FILTER_CONFIGMAP_ONLY = new KafkaProtocolFilterBuilder()
+            .withNewMetadata()
+            .withName("foo")
+            .withGeneration(42L)
+            .withUid("foo-uid")
+            .endMetadata()
+            .withNewSpec()
+            .withType("org.example.MyFilter")
+            .withConfigTemplate(Map.of(
+                    "normalProp", "normalValue",
+                    "securePropB", "${configmap:my-configmap:key}"))
+            .endSpec()
+            .build();
+
+    public static final KafkaProtocolFilter FILTER_NO_REFERENTS = new KafkaProtocolFilterBuilder()
+            .withNewMetadata()
+            .withName("foo")
+            .withGeneration(42L)
+            .withUid("foo-uid")
+            .endMetadata()
+            .withNewSpec()
+            .withType("org.example.MyFilter")
+            .withConfigTemplate(Map.of("normalProp", "normalValue"))
+            .endSpec()
+            .build();
+
     public static final Secret SECRET = new SecretBuilder()
             .withNewMetadata()
                 .withName("my-secret")
-                .withGeneration(42L)
+                .withResourceVersion("dfs")
```

Comment:
is `resourceVersion` really a string?

Reply:
yes but in the real thing it's a number like `"6266"`, I'll make it more realistic looking